src/integrationTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/CustomAccountRetrievalTest.java

```diff
@@ -106,22 +108,24 @@ void testMiAccountDataRequestSuccess() throws Exception {
                 }
             );
 
-        String createdUserId = mappedResponse.get(CreationEnum.CREATED_ACCOUNTS).get(0).toString();
+        String createdUserId = mappedResponse.get(CreationEnum.CREATED_ACCOUNTS).getFirst().toString();
 
         MockHttpServletRequestBuilder request = MockMvcRequestBuilders
             .get(MI_REPORTING_ACCOUNT_DATA_URL);
 
-        String responseMiData = mockMvc.perform(request).andExpect(status().isOk()).andReturn()
-            .getResponse().getContentAsString();
+        MvcResult responseMiData = mockMvc.perform(request).andExpect(status().isOk()).andReturn();
 
-        assertEquals(EXPECTED_HEADERS, responseMiData.split("\n")[0],
-                     "Should successfully retrieve MI data headers"
-        );
+        assertNotNull(responseMiData.getResponse(), VALIDATION_MI_REPORT);
 
-        assertTrue(
-            responseMiData.contains(createdUserId),
-            "Should successfully retrieve MI data"
+        List<AccountMiData> miData = Arrays.asList(
+            OBJECT_MAPPER.readValue(
+                responseMiData.getResponse().getContentAsString(),
+                AccountMiData[].class
+            )
         );
+
+        assertEquals(6, miData.size(), VALIDATION_MI_REPORT);
```

Comment:
Don't understand why is this has a size 6. Does it reply on the data from other tests in this class

Reply:
This is an error on my part - I added it as part of debugging for the test below and forgot to remove it. Have removed now :)